### PR TITLE
prometheus-cpp: Bump to 0.12.3 (and libcurl dependency to 7.77.0)

### PR DESCRIPTION
**Pull request by alejandro-colomar**
Specify library name and version:  **prometheus-cpp/0.12.3**

Package latest upstream source

I haven't done any tests, apart from checking the checksums with `sha256sum` (for both the version that I'm bumping, and the latest version that you already had, to make sure that I was doing it correctly).  I specifically haven't tested that this commit build correctly the new version and that it is usable, because I got an error message when trying to do so:

https://github.com/conan-io/hooks/issues/157

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by alejandro-colomar** [comment]
Hi, could someone help?  I don't understand what's going on.  There are no pkg-conf (.pc) files in the uppstream code:

```
$ tar xf v0.12.3.tar.gz
$ find prometheus-cpp-0.12.3/ | grep '\.pc$'
$
```

**Comment by alejandro-colomar** [comment]
Ahhh, upstream is now generating pkg-conf files.

How should I proceed?  Add a patch to revert that?

**Comment by alejandro-colomar** [comment]
@Fefer-Ivan , do you know what I should do to fix this?

**Comment by Fefer-Ivan** [comment]
@Alberto-Izquierdo
If you go to All logs link and search for failed package, you can download build logs.

The error is 
```[HOOK - conan-center.py] post_package(): ERROR: [PC-FILES (KB-H020)] The conan-center repository doesn't allow the packages to contain `pc` files. The packages have to be located using generators and the declared `cpp_info` information (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H020) 
[HOOK - conan-center.py] post_package(): ERROR: [PC-FILES (KB-H020)] Found files: ./lib/pkgconfig/prometheus-cpp-core.pc; ./lib/pkgconfig/prometheus-cpp-pull.pc; ./lib/pkgconfig/prometheus-cpp-push.pc (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H020) ```

Looks like prometheus-cpp added pkg-config files in later versions, so you have to delete them in recipe script like we are already deleting cmake files here: https://github.com/conan-io/conan-center-index/blob/master/recipes/prometheus-cpp/all/conanfile.py#L85

**Review by ericLemanissier — DISMISSED** [review]

**Comment by SpaceIm** [comment]
Please explicitly add pkg_config name of each component.

**Comment by alejandro-colomar** [comment]
> Please explicitly add pkg_config name of each component.

Sorry, I don't know them.  cmake autogenerates them, so they are not in the source code.  And I don't know enough of cmake to know how it generates them...

Maybe in the logs you can see those?  I don't know.

Furthermore, I can't build this locally, because it gives me some error, so I can't inspect the tree on my machine.

If you tell me what I should change, I'll do it.

Thanks,

Alex

**Comment by alejandro-colomar** [comment]
Furthermore, I followed existing conventions of your project:

```
$ grep -r tools.rm.*pkgconfig | sed 's/^[^:]*://' | uniq
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "gstreamer-1.0", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
                tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder,"lib","pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))        
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, os.path.join("lib", "pkgconfig")))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self._datadir, "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

Do you want to specify anyway?

**Comment by SpaceIm** [comment]
Below there respective `cmake_find_package_multi` names in `package_info()`, just add:

`self.cpp_info.components["prometheus-cpp-core"].names["pkg_config"] = "prometheus-cpp-core"`

`self.cpp_info.components["prometheus-cpp-push"].names["pkg_config"] = "prometheus-cpp-push"`

`self.cpp_info.components["prometheus-cpp-pull"].names["cmake_find_package_multi"] = "prometheus-cpp-pull"`

File: recipes/prometheus-cpp/all/conanfile.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ def package(self):
         cmake = self._configure_cmake()
         cmake.install()
         tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "prometheus-cpp"
```
Comment (prince-chrismc) [inline recipes/prometheus-cpp/all/conanfile.py:89]: Missing come details on the components

```
[HOOK - conan-center.py] post_package(): ERROR: [PC-FILES (KB-H020)] The conan-center repository doesn't allow the packages to contain `pc` files. The packages have to be located using generators and the declared `cpp_info` information (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H020) 
[HOOK - conan-center.py] post_package(): ERROR: [PC-FILES (KB-H020)] Found files: ./lib/pkgconfig/prometheus-cpp-core.pc; ./lib/pkgconfig/prometheus-cpp-pull.pc; ./lib/pkgconfig/prometheus-cpp-push.pc (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H020) 
ERROR: [HOOK - conan-center.py] post_package(): Some checks failed running the hook, check the output
```

At a minimum please add a fix me

```suggestion
        # FIXME: Missing prometheus-cpp-core.pc, prometheus-cpp-pull.pc prometheus-cpp-push.pc
        self.cpp_info.names["cmake_find_package"] = "prometheus-cpp"
```

You can easily add these in

Example after L93

```py
self.cpp_info.components["prometheus-cpp-core"].names["pkg_config"] = "prometheus-cpp-core"
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]

**Comment by SpaceIm** [comment]
You can also take advantage of this PR to bump libcurl version, it avoids to create a new revision for a tiny modification.

**Comment by alejandro-colomar** [comment]
After some time comparing with other recipes I got what you want :p

However, I don't really understand what it really does.  Could you please explain what those lines do to the package?  I'm curious ;)

> You can also take advantage of this PR to bump libcurl version, it avoids to create a new revision for a tiny modification.

Okay :-)

Wait for a few minutes and I'll push everything.

**Comment by SpaceIm** [comment]
> However, I don't really understand what it really does. Could you please explain what those lines do to the package? I'm curious ;)

It allows to generate proper pkgconfig files if consumers use `pkg_config` generator.

**Comment by alejandro-colomar** [comment]
> It allows to generate proper pkgconfig files if consumers use `pkg_config` generator.

So you remove the upstream ones, and then regenerate them from your recipe, right?

**Review by SpaceIm — APPROVED** [review]

**Comment by SpaceIm** [comment]
> So you remove the upstream ones, and then regenerate them from your recipe, right?

Yes, please read https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-are-cmake-findconfig-files-and-pkg-config-files-not-packaged

**Review by SpaceIm — DISMISSED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by alejandro-colomar** [comment]
I closed it temporarily, waiting for the PR regarding libcurl.  Given libcurl 7.77.0 is all green, I'll delay this one a bit, and bump the dependency of libcurl to 7.77.0 when it is available.

**Comment by alejandro-colomar** [comment]
I can't reopen it.  Could you please do it for me?
I already pushed the changes to bump libcurl to 7.77.0

**Comment by prince-chrismc** [comment]
If you can't reopen then it because you forced pushed most often

**Comment by alejandro-colomar** [comment]
Ahh, I forced pushed just before opening to avoid the CI system start a build that was going to be superseded in a moment.

I'll try to force push back, try to reopen then, and then force push again

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**